### PR TITLE
ARROW-7735: [Release][Python] Use pip to install dependencies for wheel verification

### PR DESCRIPTION
**Pull request by kszucs**
The wheel verification script fails for python 3.5. 
At the same time the wheel properly works for python 3.5 docker
images without conda environment.

Conda forge doesn't maintain packages for python 3.5 anymore
and something must have mixed with the numpy versions.

This change fixed the wheel verification locally for me.

### CODE REVIEW

**Comment by kou** [comment]
Python 3.5 + manylinux1 passes with this change.

But Python 3.5 + manylinux2000 shows the following error:

```text
+ for ml_spec in ${manylinuxes}
+ [[ 3.5m = \2\.\7\m\u ]]
+ pip install python-rc/0.16.0-rc2/pyarrow-0.16.0-cp35-cp35m-manylinux2010_x86_64.whl
pyarrow-0.16.0-cp35-cp35m-manylinux2010_x86_64.whl is not a supported wheel on this platform.
```

```console
$ python -c "from wheel import pep425tags; print('\n'.join(['-'.join(t) for t in pep425tags.get_supported()]))"
cp35-cp35m-linux_x86_64
cp35-abi3-linux_x86_64
cp35-none-linux_x86_64
cp34-abi3-linux_x86_64
cp33-abi3-linux_x86_64
cp32-abi3-linux_x86_64
cp35-none-any
cp3-none-any
cp34-none-any
cp33-none-any
cp32-none-any
cp31-none-any
cp30-none-any
py3-none-linux_x86_64
py35-none-any
py3-none-any
py34-none-any
py33-none-any
py32-none-any
py31-none-any
py30-none-any
```

**Comment by kou** [comment]
It may be better that we use Docker (`python:2.7`, `python:3.5`, `python:3.6` and `python:3.7`) instead of Conda.

```console
$ % docker run -it --rm python:3.5
Python 3.5.9 (default, Feb  2 2020, 10:14:19) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pip._internal import pep425tags; print('\n'.join([str(t) for t in pep425tags.get_supported()]))
cp35-cp35m-manylinux2014_x86_64
cp35-cp35m-manylinux2010_x86_64
cp35-cp35m-manylinux1_x86_64
cp35-cp35m-linux_x86_64
cp35-abi3-manylinux2014_x86_64
cp35-abi3-manylinux2010_x86_64
cp35-abi3-manylinux1_x86_64
cp35-abi3-linux_x86_64
cp35-none-manylinux2014_x86_64
cp35-none-manylinux2010_x86_64
cp35-none-manylinux1_x86_64
cp35-none-linux_x86_64
cp34-abi3-manylinux2014_x86_64
cp34-abi3-manylinux2010_x86_64
cp34-abi3-manylinux1_x86_64
cp34-abi3-linux_x86_64
cp33-abi3-manylinux2014_x86_64
cp33-abi3-manylinux2010_x86_64
cp33-abi3-manylinux1_x86_64
cp33-abi3-linux_x86_64
cp32-abi3-manylinux2014_x86_64
cp32-abi3-manylinux2010_x86_64
cp32-abi3-manylinux1_x86_64
cp32-abi3-linux_x86_64
py35-none-manylinux2014_x86_64
py35-none-manylinux2010_x86_64
py35-none-manylinux1_x86_64
py35-none-linux_x86_64
py3-none-manylinux2014_x86_64
py3-none-manylinux2010_x86_64
py3-none-manylinux1_x86_64
py3-none-linux_x86_64
py34-none-manylinux2014_x86_64
py34-none-manylinux2010_x86_64
py34-none-manylinux1_x86_64
py34-none-linux_x86_64
py33-none-manylinux2014_x86_64
py33-none-manylinux2010_x86_64
py33-none-manylinux1_x86_64
py33-none-linux_x86_64
py32-none-manylinux2014_x86_64
py32-none-manylinux2010_x86_64
py32-none-manylinux1_x86_64
py32-none-linux_x86_64
py31-none-manylinux2014_x86_64
py31-none-manylinux2010_x86_64
py31-none-manylinux1_x86_64
py31-none-linux_x86_64
py30-none-manylinux2014_x86_64
py30-none-manylinux2010_x86_64
py30-none-manylinux1_x86_64
py30-none-linux_x86_64
cp35-none-any
py35-none-any
py3-none-any
py34-none-any
py33-none-any
py32-none-any
py31-none-any
py30-none-any
```

**Comment by kszucs** [comment]
@kou for testing the produced wheels [we already use](https://github.com/apache/arrow/blob/master/dev/tasks/tasks.yml#L642) these docker images - which doesn't mean that we cannot rerun these checks during the release verification process.

Although docker would only be suitable for testing the linux wheels.

**Comment by pitrou** [comment]
> But Python 3.5 + manylinux2000 shows the following error:

It's quite possible that manylinux2010 doesn't have tooling for Python 3.5.
We should stop caring about Python 3.5. It's [obsolete](ARROW-5679).

File: dev/release/verify-release-candidate.sh [hunk 1]
```diff
@@ -609,9 +609,11 @@ test_linux_wheels() {
...
       check_python_imports py_arch
 
-      # execute the python unit tests
-      conda install -y --file ${ARROW_DIR}/ci/conda_env_python.yml pandas
-      pytest --pyargs pyarrow
+      # install test requirements
+      pip install -r ${ARROW_DIR}/python/requirements-test.txt
+
+      # execute the tests
+      pytest -sv --pyargs pyarrow
```
Comment (pitrou) [inline dev/release/verify-release-candidate.sh]: The `-s` and `-v` flags are spammy. Better to remove them IMHO.

**Review by pitrou — COMMENTED** [review]
+1

File: dev/release/verify-release-candidate.sh (hunk 1 above)
Comment (kszucs) [reply dev/release/verify-release-candidate.sh]: Just a sec

**Review by kszucs — COMMENTED** [review]
+1

**Pull request closed without merging**